src/common.c

```diff
@@ -337,6 +343,19 @@ static int ofi_str_to_psmx(const char *str, void **addr, size_t *len)
 	return (ret == 1) ? 0 : -FI_EINVAL;
 }
 
+static int ofi_str_to_psmx2(const char *str, void **addr, size_t *len)
+{
+	int ret;
+
+	*len = 2 * sizeof(uint64_t);
+	if (!(*addr = calloc(1, *len)))
+		return -FI_ENOMEM;
+
+	ret = sscanf(str, "%*[^:]://%" SCNu64 ":%" SCNu64,
+		     (uint64_t *) *addr, (uint64_t *) *addr + 1);
+	return (ret == 2) ? 0 : -FI_EINVAL;
```

Comment:
We should probably free addr if we return a failure.  I'm guessing I have this error in the other str_to_xxx calls as well.